js/web/lib/wasm/jsep/backend-webnn.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Current version of the WebNN API does not have a TypeScript definition file.
+// https://github.com/webmachinelearning/webnn/issues/677
+/// <reference path="./webnn/webnn.d.ts" />
+
+
+class ContextState {
+    context: MLContext;
+    sessions: Set<number>;
+}
+
+export class WebNNBackend {
+    contextStates: Map<string, ContextState>
+    contextsBySessionId: Map<number, string>
+
+    constructor() {
+        this.contextStates = new Map();
+        this.contextsBySessionId = new Map();
+    }
+
+    async createContext(options: MLContextOptions): Promise<string> {
+        const contextId = JSON.stringify(options);
```

Comment:
`numThreads` is cpu only option, shall we ignore it for npu and gpu devices?  Thus we can reduce noise, e.g. in case user creating two gpu sessions with different `numThreads`, although it is rare.

Reply:
I have removed `numThreads` when the device type is not `cpu` from they key. I have also added sorting to the object keys.